docs/aws/audit/opensearchmonitoring/rules/opensearch_access_control_enabled_remediation.mdx

```diff
@@ -1,6 +1,240 @@
+
+   ```sh
+   aws opensearch update-domain-config --domain-name my-domain \
+       --advanced-security-options Options={Enabled=true,InternalUserDatabaseEnabled=true,MasterUserOptions={MasterUserName=admin,MasterUserPassword=AdminPassword123}}
+   ```
+
+3. **Configure Access Policies:**
+   Ensure that your domain has the correct access policies to support Fine-Grained Access Control. This can be done using the `update-domain-config` command to set the access policies.
+
+   ```sh
+   aws opensearch update-domain-config --domain-name my-domain \
+       --access-policies '{
+           "Version": "2012-10-17",
+           "Statement": [
+               {
+                   "Effect": "Allow",
+                   "Principal": {
+                       "AWS": "*"
+                   },
+                   "Action": "es:*",
+                   "Resource": "arn:aws:es:region:account-id:domain/my-domain/*"
+               }
+           ]
+       }'
+   ```
+
+4. **Verify Fine-Grained Access Control Configuration:**
+   After enabling Fine-Grained Access Control, verify the configuration to ensure it is correctly applied.
+
+   ```sh
+   aws opensearch describe-domain --domain-name my-domain
+   ```
+
+   Look for the `AdvancedSecurityOptions` in the output to confirm that Fine-Grained Access Control is enabled.
+
+By following these steps, you can ensure that Fine-Grained Access Control is enabled for your OpenSearch Service Domains using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of not having Fine-Grained Access Control enabled in OpenSearch Service Domains using Python scripts, you can follow these steps:
+
+1. **Install AWS SDK for Python (Boto3):**
+   Ensure you have Boto3 installed in your environment. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials:**
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables:
+   ```bash
+   aws configure
+   ```
+
+3. **Create a Python Script to Enable Fine-Grained Access Control:**
+   Write a Python script that uses Boto3 to enable Fine-Grained Access Control for your OpenSearch Service Domain.
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon OpenSearch Service
+   client = boto3.client('opensearch')
+
+   # Define the domain name
+   domain_name = 'your-domain-name'
+
+   # Enable Fine-Grained Access Control
+   response = client.update_domain_config(
+       DomainName=domain_name,
+       AdvancedSecurityOptions={
+           'Enabled': True,
+           'InternalUserDatabaseEnabled': True,
+           'MasterUserOptions': {
+               'MasterUserARN': 'arn:aws:iam::123456789012:user/master-user',
+               'MasterUserName': 'master-username',
+               'MasterUserPassword': 'master-password'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>